starter_code/src/movies.js

```diff
@@ -1,20 +1,2095 @@
+    genre: ['Crime', 'Drama', 'Mystery', 'Romance', 'Thriller'],
+    rate: '8.1'
+  },
+  {
+    title: 'Pirates of the Caribbean: The Curse of the Black Pearl',
+    year: '2003',
+    director: 'Gore Verbinski',
+    duration: '2h 23min',
+    genre: ['Action', 'Adventure', 'Fantasy'],
+    rate: '8.0'
+  },
+  {
+    title: 'PK',
+    year: '2014',
+    director: 'Rajkumar Hirani',
+    duration: '2h 33min',
+    genre: ['Comedy', 'Drama', 'Fantasy', 'Sci-Fi'],
+    rate: '8.2'
+  },
+  {
+    title: 'Dog Day Afternoon',
+    year: '1975',
+    director: 'Sidney Lumet',
+    duration: '2h 5min',
+    genre: ['Biography', 'Crime', 'Drama', 'Thriller'],
+    rate: '8.0'
+  },
+  {
+    title: 'Dead Poets Society',
+    year: '1989',
+    director: 'Peter Weir',
+    duration: '2h 8min',
+    genre: ['Comedy', 'Drama'],
+    rate: '8.0'
+  }
+];
 
 
-// Get the average of all rates with 2 decimals 
 
+// ITERATION 1
+function turnHoursToMinutes(array){
+  return array.map(function(element){
+    var movie = Object.assign({}, element);
+    var duration = movie.duration.split(' ').reduce(function(acc, curr){
+      if (curr.indexOf('h') !== -1) {
+        return Number(curr.replace('h', '')) * 60 + acc;
+      } else{
+        return Number(curr.replace('min', '')) + acc;
+      }
+    }, 0);
+    movie.duration = duration;    
+    return movie;
+  }); 
 
-// Get the average of Drama Movies
+} 
+console.log(turnHoursToMinutes(movies));
 
 
-// Order by time duration, in growing order
+////ITERATION 2 AVERAGE RATE
+function ratesAverage(array){
+  return (array.reduce(function(acc, curr){    
+    return Number(acc + Number(curr.rate));
+  },0) / array.length).toFixed(2);
+}
+console.log(ratesAverage(movies));
 
 
-// How many movies did STEVEN SPIELBERG
+////ITERATION 3 RE-USE ANOTHER FUNCTION
+function dramaMoviesRate(array){
+  var arrayDrama =  array.filter(function(elem){    
+    return elem.genre.indexOf('Drama') !== -1;
+  });
+  return ratesAverage(arrayDrama);
+}
+console.log(dramaMoviesRate(movies));
 
+///ITERATION 3 LO PONGO DETRAS PERO SI TENGO FUNCIONES SEPARADAS
+function dramaMoviesRate2(array){
+  var arrayDrama =  array.filter(function(el){
+    return el.genre.indexOf('Drama') !== -1;
+  });
 
-// Order by title and print the first 20 titles
+  return (arrayDrama.reduce(function(acc, curr){
+    return acc + Number(curr.rate);
+  }, 0) / arrayDrama.length).toFixed(2);
+}
+console.log(dramaMoviesRate2(movies));
 
 
-// Best yearly rate average
+///ITERATION 3 LO HAGO TODO JUNTO PERO HASTA QUE NO TERMINE EL BUCLE NO HAGO LA DIVISION POR ESO EL IF
+function dramaMoviesRate3(array){
+  return array.filter(function(elem, index){     
+    return elem.genre.indexOf('Drama') !== -1;
+  }).reduce(function(acc, elem, index, arrayDrama){
+    if (index === arrayDrama.length -1) {            
+      return ((acc + Number(elem.rate)) / arrayDrama.length).toFixed(2);
+    } else{
+      return acc + Number(elem.rate);
+    }
+  },0);
+}
```

Comment:
No hagáis 3 veces la misma función, hacedla una sola vez, Jasmine tampoco lo pasa.